### PR TITLE
schedule: fix timeline garbage collection

### PR DESCRIPTION
**Pull request by bmr-cymru**
The Timeline garbage collection policy has a bug (#592) where the first snapshot set created by a policy can be immediately deleted, even when the user has configured retention for weekly, daily, or hourly snapshots.

This happens because the GcPolicyParamsTimeline class currently classifies snapshot sets by greedily assigning to categories "yearly", "quarterly", "monthly", "weekly", "daily", "hourly" - meaning that the first snapshot set is always classified as "yearly", and hence is subject to deletion if keep_yearly=0.

* Fix this by allowing snapshot sets to be assigned to multiple matching categories as appropriate (i.e. a snapshot set taken at Midnight, Mon 1 Jan 2024 is simultaneously "yearly", "weekly", "daily", and "hourly"), and only assigning a snapshot set for deletion if all category retention policies yield False (#595).

* Correct a mis-calculation in the COUNT policy when `keey_count > len(sets)`

* Additionally, implement a new debug subsystem, "schedule", add it to `--debug`, and add debug logging to each `GcPolicyParams` subclass's `evaluate()` method. (#593, #594) and finally, as a minor UX improvement make `-c|--config` mandatory for `gc` and `create-scheduled` commands (#598).

Resolves: #592
Resolves: #593
Resolves: #594
Resolves: #595
Resolves: #597
Resolves: #598
Resolves: #600

## Summary by CodeRabbit

* **Bug Fixes**
  * Timeline retention now treats multi‑category snapshots conservatively — a snapshot is kept if any category retains it; first/edge‑case behaviour and expected deletions adjusted.

* **New Features**
  * Added a schedule debug option and richer schedule‑subsystem debug logging.
  * Garbage collection now reports and prints a summary list of deleted snapshot sets.
  * Schedule command now requires a config argument.

* **Tests**
  * Added regression tests for first‑snapshot edge cases, Jan‑1 weekday, multi‑category retention and multiple weekly snapshots.

* **Documentation**
  * Manpage synopsis updated to require schedule arguments.

### CODE REVIEW

**Comment by bmr-cymru** [comment]
Note: `test_GcPolicyParamsTimeline_evaluate_2` in the `container_tests` suite is currently failing with an off-by-one:

```text
FAILED tests/test_schedule.py::GcPolicyTests::test_GcPolicyParamsTimeline_evaluate_2 - AssertionError: 17 != 18
```

This needs further investigation to determine if it's a bug in the evaluation or the test as a result of the multi-categorisation change.

File: snapm/_snapm.py [hunk 1]
```diff
@@ -33,21 +33,28 @@
...
+SNAPM_DEBUG_SCHEDULE = 8
+SNAPM_DEBUG_MOUNTS = 16
 SNAPM_DEBUG_ALL = (
-    SNAPM_DEBUG_MANAGER | SNAPM_DEBUG_COMMAND | SNAPM_DEBUG_REPORT | SNAPM_DEBUG_MOUNTS
+    SNAPM_DEBUG_MANAGER
+    | SNAPM_DEBUG_COMMAND
+    | SNAPM_DEBUG_REPORT
+    | SNAPM_DEBUG_SCHEDULE
+    | SNAPM_DEBUG_MOUNTS
 )
```
[the remark this answers is not in the document]
Comment (bmr-cymru) [reply snapm/_snapm.py]: @coderabbitai: Nobody persists the numeric mask values. If anything did it would be a huge bug. Taking your example, `SNAPM_DEBUG_ALL==15` would break the meaning of "_ALL" in that context, since it would reflect only subsystems that existed at the time it was written, where the value of "SNAPM_DEBUG_ALL" changes over time as new subsystems are added. #notabug.

**Comment by bmr-cymru** [comment]
@coderabbitai: full review
